Monika After Story/game/script-songs.rpy

```diff
@@ -741,3 +741,206 @@ label mas_song_hero_long:
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="mas_monika_plays_yr",
+            category=['monika','music'],
+            prompt="Can you play 'Your Reality' for me?",
+            unlocked=False,
+            pool=True,
+            rules={"no unlock": None}
+        )
+    )
+
+label mas_monika_plays_yr(skip_leadin=False):
+    if not skip_leadin:
+        if not renpy.seen_audio(songs.FP_YOURE_REAL) and not persistent.monika_kill:
+            m 2eksdlb "Oh, ahaha! You want me to play the original version, [player]?"
+            m 2eka "Even though I've never played it for you, I suppose you've heard it on the soundtrack or saw it on youtube, huh?"
+            m 2hub "The ending isn't my favorite, but I'll still be happy to play it for you!"
+            m 2eua "Just let me get the piano.{w=0.5}.{w=0.5}.{nw}"
+
+        else:
+            m 3eua "Sure, let me just get the piano.{w=0.5}.{w=0.5}.{nw}"
+
+    window hide
+    $ HKBHideButtons()
+    $ mas_RaiseShield_core()
+    $ store.songs.enabled = False
+
+    show monika at rs32
+    hide monika
+    pause 3.0
+    show mas_piano at lps32,rps32 zorder MAS_MONIKA_Z+1
+    pause 5.0
+    show monika at ls32 zorder MAS_MONIKA_Z
+    show monika 6dsa
+    pause 2.0
+    $ play_song(store.songs.FP_YOURE_REAL,loop=False)
+
+    show monika 6hua
+    $ renpy.pause(10.012)
+    show monika 6eua_static
+    $ renpy.pause(5.148)
+    show monika 6hua
+    $ renpy.pause(3.977)
+    show monika 6eua_static
+    $ renpy.pause(5.166)
+    show monika 6hua
+    $ renpy.pause(3.743)
+    show monika 6esa
+    $ renpy.pause(9.196)
+    show monika 6eka
+    $ renpy.pause(13.605)
+    show monika 6dua
+    $ renpy.pause(9.437)
+    show monika 6eua_static
+    $ renpy.pause(5.171)
+    show monika 6dua
+    $ renpy.pause(3.923)
+    show monika 6eua_static
+    $ renpy.pause(5.194)
+    show monika 6dua
+    $ renpy.pause(3.707)
+    show monika 6eka
+    $ renpy.pause(16.884)
+    show monika 6dua
+    $ renpy.pause(20.545)
+    show monika 6eka_static
+    $ renpy.pause(4.859)
+    show monika 6dka
+    $ renpy.pause(4.296)
+    show monika 6eka_static
+    $ renpy.pause(5.157)
+    show monika 6dua
+    $ renpy.pause(8.064)
+    show monika 6eka
+    $ renpy.pause(22.196)
+    show monika 6dka
+    $ renpy.pause(3.630)
+    show monika 6eka_static
+    $ renpy.pause(1.418)
+    show monika 6dka
+    $ renpy.pause(9.425)
+    show monika 5dka with dissolve
+    $ renpy.pause(5)
+
+    stop music
+    show monika 6eua at rs32 with dissolve
+    pause 1.0
+    hide monika
+    pause 3.0
+    hide mas_piano
+    pause 6.0
+    show monika 1eua at ls32 zorder MAS_MONIKA_Z
+
+    if not skip_leadin:
+        if store.mas_globals.dlg_workflow:
+            $ mas_resetTextSpeed()
+            $ mas_MUMUDropShield()
+            $ enable_esc()
+        else:
+            $ mas_DropShield_core()
```

Comment:
what are the workflows that trigger this or `plays_or`? If its all from `call_next_event`, then we don't need to check dlg_workflow and should always assume we are in a dialogue workflow.